components/NavBar.tsx

```diff
@@ -0,0 +1,35 @@
+import { AppBar, Tabs, Tab, Toolbar } from "@mui/material";
+import React from "react";
+import { Router, withRouter } from "next/dist/client/router";
+
+class NavBar extends React.Component<{ router: Router }> {
+  handleChange(path: string): void {
+    void this.props.router.push(path);
+  }
```

Comment:
Why are you using class components here? It's better to use functional components